### PR TITLE
[ROCm][CI] Re-enable ROCm CI & make CI changes

### PR DESCRIPTION
**Pull request by akashveramd**
Re-enabling ROCm CI as this PR https://github.com/pytorch/torchtitan/pull/4002 temporarily disabled ROCm CI. 

This PR also makes following CI changes-
- Separates CUDA & ROCm workflows for `integration_test_8gpu_features.yaml` and `integration_test_8gpu_models.yaml`
- Separate badging on README.md to indicate ROCm/CUDA workflow status separately
- Remove `ciflow/8gpu` label as a trigger for the CUDA-specific workflows
- Add `ciflow/rocm` label as a trigger for the ROCm-specific workflows
-- This label can be added manually by PR developers wanting to test on ROCm
-- This label will also be automatically added by the pytorch bot for any PRs that mention ROCm in the title/description
- Skipped on ROCm 7.14: exploding gradients in Qwen3 MoE TP+EP training
-- Will be re-enabled for ROCm7.15

### CODE REVIEW

**Comment by akashveramd** [comment]
@tianyu-l: I created this PR to re-enable ROCm CI. Could you please approve the workflows?

**Comment by jithunnair-amd** [comment]
@akashveramd Hmm, looks like the last run picked up a torch nightly wheel which didn't yet have the rocm-smi->amd-smi changes propagated. Can you please push an empty commit to retrigger CI so it picks up a newer nightly?

**Comment by akashveramd** [comment]
@jithunnair-amd: Pushed the dummy commit. @tianyu-l the workflow requires re-approval due to the latest changes. Could you please re-approve?

**Comment by jithunnair-amd** [comment]
@tianyu-l Please re-approve the workflows. Also, as on pytorch/pytorch repo, having the ability to approve CI workflows on PRs would be very helpful to decrease the turnaround time on PRs that attempt to put in updates/fixes for ROCm CI on torchtitan.
cc @huydhn in case he can help with permissions

**Comment by akashveramd** [comment]
@tianyu-l: I made a change to disable ROCm CI run for draft PRs. Could you please approve the workflows for this PR?
As a sanity check I am keeping the PR in draft mode and it should not run the ROCm workflows after you approve the workflows. Later I can move it out of the draft mode.

**Comment by tianyu-l** [comment]
oops I accidentally clicked "Ready for review"

**Comment by akashveramd** [comment]
@tianyu-l: All checks are passing now, and PR is ready for review.

**Review by wwwjn — COMMENTED** [review]
I have another PR #4127 which is hugely refactor the CI, and also separate ROCm and CUDA, could you rebase and enable rocm following #4127 's idea? Thank you

**Comment by jithunnair-amd** [comment]
> I have another PR #4127 which is hugely refactor the CI, and also separate ROCm and CUDA, could you rebase and enable rocm following #4127 's idea? Thank you

We will assess how this PR should change in light of PR 4127.

**Comment by akashveramd** [comment]
@tianyu-l:  Closing this PR as this PR https://github.com/pytorch/torchtitan/pull/4361 will take over.
